src/structures/GuildMember.js

```diff
@@ -284,6 +302,18 @@ class GuildMember extends Base {
     return this.edit({ nick }, reason);
   }
 
+  
+  /**
+   * Disables or enables communication for this member.
+   * @param {?Date} The duration in milliseconds the guild member should have communication disabled for, or `null` to enable communication for them
+   * @param {string} [reason] Reason for setting the nickname
+   * @returns {Promise<GuildMember>}
+   */
+  disableCommunication(duration, reason) {
+    const timestamp = Date.now()+duration;
+    return this.edit({communication_disabled_until: new Date(timestamp).toISOString()}, reason);
```

Comment:
The edit method accepts properties in camelCase. All of these checks should be done there and the duration should be a value in seconds with how long to mute the member for.

Reply:
what checks exactly?

Reply:
Everything this function is doing. It should only be a call to this.edit with the provided value.

Reply:
i moved the parsing to GuildMemberManager.edit